docs/compiler.md

```diff
@@ -50,14 +49,14 @@ Python is considered to be an accelerator and must additionally specify `shared_
 The default compilers can provide examples compatible with your system once Pyccel has been executed at least. To export the JSON file describing your setup, use the `--export-compile-info` flag and provide a target file name.
 E.g.
 ```shell
-pyccel --compiler=intel --language=c --export-compile-info=icc.json
+pyccel --compiler=intel --export-compile-info=intel.json
 ```
 once this file has been modified it can then be used with:
 ```shell
-pyccel --compiler=icc.json --language=c <file_to_translate>
+pyccel --compiler=intel.json <file_to_translate>
```

Comment:
I think we should mention that the environment variable `PYCCEL_DEFAULT_COMPILER` can be used to specify the path to the JSON file. This is not obvious just by reading the section "Specifying a compiler".

Reply:
Better : 40af93c24 ?

Reply:
Do you want me to do it?

Reply:
If you have a suggestion then feel free to add it :+1:

Reply:
The flags should override the environment variable

https://github.com/pyccel/pyccel/blob/bbff872ce78c57d2c91d1895ca154ba8fe3b7d1a/pyccel/codegen/pipeline.py#L186-L187

Reply:
Thanks, I will mention this

Reply:
What do you think? [d23bed8](https://github.com/pyccel/pyccel/pull/2252/commits/d23bed8d330b848d8453e6b355f4f034497fbebb)

Reply:
I think I misunderstood your question about flags. I thought you were talking about `--X` options not the `--flags` option. A compiler definition does not contain a `flags` field. It has `general_flags`, `release_flags` and `debug_flags`. The flags used in the compilations are `general_flags`, one of `release_flags` and `debug_flags`, and any additional flags passed using the `--flags` option.

Reply:
No, you understood correctly what I meant. I was talking about any option that is also contained in the JSON file. But I was sloppy in my explanation, as you rightly pointed out things are more complicated

Reply:
Corrected 
[583c6f6](https://github.com/pyccel/pyccel/pull/2252/commits/583c6f6b1f6e39468ed4bab4e8b58331998694b2)

Reply:
Thanks!